### PR TITLE
Fix to slow perf of PipeBlob uploads 

### PR DESCRIPTION
**Pull request by wonwuakpa-msft**
Assigned the pipingUploadParallelism to the user defined environment variable AZCOPY_CONCURRENCY_VALUE if present. If not, fell back on the default concurrency value and errored out for AUTO and non-integer values.

### CODE REVIEW

**Review by adreed-msft — APPROVED** [review]
Pair-coded on this

File: README.md [hunk 1]
```diff
@@ -1,4 +1,7 @@
 # AzCopy v10
+wendi
```
Comment (adreed-msft) [inline README.md:2]: should probably remove these lines though :)

File: README.md [hunk 2]
```diff
@@ -1,5 +1,4 @@
 # AzCopy v10
-
```
Comment (dphulkar-msft) [inline README.md:2]: Is this change required? If not can we remove this?

File: cmd/copy.go [hunk 3]
```diff
@@ -1347,6 +1348,27 @@ func (cca *CookedCopyCmdArgs) processRedirectionDownload(blobResource common.Res
 func (cca *CookedCopyCmdArgs) processRedirectionUpload(blobResource common.ResourceString, blockSize int64) error {
 	ctx := context.WithValue(context.TODO(), ste.ServiceAPIVersionOverride, ste.DefaultServiceApiVersion)
 
+	// get the concurrency environment value
+	userDefnConcurrencyStr := glcm.GetEnvironmentVariable(common.EEnvironmentVariable.ConcurrencyValue())
+
+	pipingUploadParallelism := pipingUploadParallelism
```
Comment (dphulkar-msft) [inline cmd/copy.go:1354]: Could you rename the local variable as its looks confusing at first glance?

File: cmd/copy.go [hunk 4]
```diff
@@ -1347,6 +1348,27 @@ func (cca *CookedCopyCmdArgs) processRedirectionDownload(blobResource common.Res
...
+		if userDefnConcurrencyStr == "AUTO" {
+			return errors.New("concurrency auto-tuning is not possible when using redirection transfers (AZCOPY_CONCURRENCY_VALUE = AUTO)")
+		}
+
+		// convert the concurrency value to int
+		concurrencyValue, err := strconv.ParseInt(userDefnConcurrencyStr, 10, 32)
+
+		//handle the error if the conversion fails
+		if err != nil {
+			return fmt.Errorf("AZCOPY_CONCURRENCY_VALUE is not set to a valid value, an integer is expected (current value: %s): %s", userDefnConcurrencyStr, err.Error())
```
Comment (dphulkar-msft) [inline cmd/copy.go:1366]: instead of using err.Error(), you can directly pass err to fmt.Errorf, as it implements the error interface.
Also you can use %w verb for error wrapping in Go, which makes it easier to inspect errors later.

File: cmd/copy.go (hunk 3 above)
Comment (wonwuakpa-msft) [reply cmd/copy.go:1354]: Changed it to `concurrencyEnvVar`

**Review by vibhansa-msft — APPROVED** [review]

**Comment by dphulkar-msft** [comment]
LGTM. Please resolve the conflicts.

**Review by dphulkar-msft — APPROVED** [review]
